soroban-sdk/src/storage.rs

```diff
@@ -299,17 +296,6 @@ impl Storage {
     fn get_internal(&self, key: Val, storage_type: StorageType) -> Val {
         internal::Env::get_contract_data(&self.env, key, storage_type).unwrap_infallible()
     }
-
-    #[cfg(any(test, feature = "testutils"))]
-    fn verify_in_contract(&self, storage_kind: &str) {
-        if !self.env.in_contract() {
-            panic!(
-                "'{storage_kind}' storage is not accessible outside of the \
-                    contract, wrap the call with `env.as_contract()` to \
-                    access it from a particular contract"
-            );
-        }
-    }
```

Comment:
@dmkozh Hope you don't mind I took the change you pushed in #1288 and repurposed it for use across the SDK as a quick assertion drop in macro. I changed it to a macro so the assertion will render as occurring at the location of use and generalised the error message to find a balance between being as useful to being easy to drop around the SDK.

Reply:
That's indeed nicer, thanks.